### DISCUSSION TITLE
Audiobooks

### DISCUSSION THREAD

**Discussion by Melechtna** [Kareadita/Kavita#2923] [Ideas]
### Idea Description

This really feels like the one and only thing Kavita is missing for me to be a "feature complete" ebook handler to where I could basically ditch all my other software for handling ebooks, which I personally would count audiobooks as. I don't even think it would necessarily be that difficult versus what's already here, a simple audio library, a player, and adding a timestamp to the database similar to the page number, maybe some chaptering stuff for audiobook titles that support it, play, pause, skip chapter forward and back buttons, and a seek bar, and there you go. Give them their own category, and I see no reason why this wouldn't basically drop into the existing interface.

### Idea Category

User Experience

### Duration of Using Kavita

About a month now

### Before submitting

- [X] I've already searched for existing ideas before posting.

**Comment by therobbiedavis** [comment]
Joe has stated in the past this is not something he is interested in, however he is open to PRs.

**Reply by Melechtna** [reply]
Well, not making any promises, but I do have some web development experience. I might could see about doing this in my free time.

**Reply by maximo1984** [reply]
I would also like to get into modding some stuff. I have some coding experience, but not github. It has always been confusing to me. How would I go about altering the code?

**Reply by majora2007** [reply]
You fork the project, clone, modify, push your branch up, then PR into mainline.

**Reply by Melechtna** [reply]
To add onto that, there's two ways you can handle this, either with github desktop (what I personally use) which kind of automates the process and (IMO) makes the process easier. Or you can use the cli tools available for your OS. If you're using Linux, these come baked in, anything else, you have to go get it.

**Comment by LeviBickel** [comment]
I tried the pull request and he wasn't open to including it. I realize I am late to this thread but I created a fork if you end up needing it [here](https://github.com/LeviBickel/Kavita)
